sklearn/tree/_classes.py

```diff
@@ -163,6 +162,7 @@ def fit(self, X, y, sample_weight=None, check_input=True,
 
         # Determine output settings
         n_samples, self.n_features_ = X.shape
+        self.n_features_in_ = self.n_features_
```

Comment:
Shall we deprecate the `n_features_` attribute then?

Reply:
Yup, I am planning on doing it in a follow up PR.